linkerd/protocol/h2/src/main/scala/io/buoyant/linkerd/protocol/H2Config.scala

```diff
@@ -255,7 +255,7 @@ case class RetryBufferSize(
 
 class H2ServerConfig extends ServerConfig with H2EndpointConfig {
 
-  var maxConcurrentStreamsPerConnection: Option[Int] = None
+  var maxConcurrentStreamsPerConnection: Option[Long] = None
```

Comment:
I think we may need to include the `@JsonDeserialize` annotation since this is a long similar to `classificationTimeoutMs`. It would look something like this:

```scala
 @JsonDeserialize(contentAs = classOf[java.lang.Long])
  var maxConcurrentStreamsPerConnection: Option[Long] = None
```

Reply:
Ah, nice catch. I'm going to revert the `Long` -> `Int` change, and handle it in `withEndpointParams`.